go-controller/pkg/util/nicstobridge.go

```diff
@@ -228,46 +275,90 @@ func NicToBridge(iface string) (string, error) {
 		return "", err
 	}
 
-	bridge := GetBridgeName(iface)
-	stdout, stderr, err := RunOVSVsctl(
-		"--", "--may-exist", "add-br", bridge,
-		"--", "br-set-external-id", bridge, "bridge-id", bridge,
-		"--", "br-set-external-id", bridge, "bridge-uplink", iface,
-		"--", "set", "bridge", bridge, "fail-mode=standalone",
-		fmt.Sprintf("other_config:hwaddr=%s", ifaceLink.Attrs().HardwareAddr),
-		"--", "--may-exist", "add-port", bridge, iface,
-		"--", "set", "port", iface, "other-config:transient=true")
+	bridge, err := createOVSBridge(iface, ifaceLink.Attrs().HardwareAddr.String())
 	if err != nil {
-		klog.Errorf("Failed to create OVS bridge, stdout: %q, stderr: %q, error: %v", stdout, stderr, err)
 		return "", err
 	}
-	klog.Infof("Successfully created OVS bridge %q", bridge)
-
 	setupDefaultFile()
 
-	// Get ip addresses and routes before any real operations.
-	family := syscall.AF_UNSPEC
-	addrs, err := netLinkOps.AddrList(ifaceLink, family)
+	bridgeLink, err := netLinkOps.LinkByName(bridge)
 	if err != nil {
 		return "", err
 	}
-	routes, err := netLinkOps.RouteList(ifaceLink, family)
-	if err != nil {
+
+	if err := saveIPConfig(ifaceLink, bridgeLink, false); err != nil {
 		return "", err
 	}
+	return bridge, nil
+}
+
+// SyncBypassConfig adds bypass representor to the bridge and moves IP addresses and
+// IP routes to bypass interface from the iface link
+func SyncBypassConfig(bridge, iface, bypassIface, bypassRep string) error {
+	if stdout, stderr, err := RunOVSVsctl("--", "--may-exist", "add-port", bridge, bypassRep); err != nil {
+		klog.Errorf("Failed to add %s interface port to OVS bridge, stdout: %q, stderr: %q, error: %v",
+			bypassRep, stdout, stderr, err)
+		return err
+	}
 
 	bridgeLink, err := netLinkOps.LinkByName(bridge)
+	if err != nil {
+		return err
+	}
+
+	ifaceLink := bridgeLink
+	moveMasqConf := false
+	if bridge != iface {
+		ifaceLink, err = netLinkOps.LinkByName(iface)
+		if err != nil {
+			return err
+		}
+	} else if config.OvnKubeNode.Mode == types.NodeModeFull {
```

Comment:
Is Bypass Port supported for DPU mode? If so, wouldn't we need to do this for the DPU case?

Reply:
Should we leave a commentary note? Or implement such case for the future?

Reply:
I had to removed this check in the next iteration, because it is irrelevant to the node mode. Whenever this use case appear in the future it should work out of the box.